core-web/apps/dotcms-ui/src/app/portlets/dot-categories/dot-categories-list/store/dot-categories-list-store.ts

```diff
@@ -0,0 +1,196 @@
+import { Injectable } from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import { MenuItem } from 'primeng/api';
+import { DataTableColumn } from '@models/data-table';
+import { DotMessageService } from '@services/dot-message/dot-messages.service';
+import { DotCategory } from '@dotcms/app/shared/models/categories/dot-categories.model';
+
+export interface DotCategoriesListState {
+    getCategoryEndPoint: string;
+    categoriesBulkActions: MenuItem[];
+    selectedCategories: DotCategory[];
+    addToBundleIdentifier: string;
+    tableColumns: DataTableColumn[];
+    categoryBreadCrumb: MenuItem[];
+    breadCrumbStarterIcon: MenuItem;
+    paginationPerPage: number;
+}
+
+@Injectable()
+export class DotCategoriesListStore extends ComponentStore<DotCategoriesListState> {
+    constructor(private dotMessageService: DotMessageService) {
+        super(null);
+        this.breadCrumbStarterIcon = { icon: 'pi pi-home', disabled: true };
+        this.setState({
+            getCategoryEndPoint: 'v1/categories',
+            categoriesBulkActions: this.getCategoriesActions(),
+            tableColumns: this.getCategoriesColumns(),
+            addToBundleIdentifier: '',
+            selectedCategories: [],
+            categoryBreadCrumb: this.getCategoryBreadCrumbs(),
+            breadCrumbStarterIcon: this.breadCrumbStarterIcon,
+            paginationPerPage: 25
+        });
```

Comment:
```suggestion
        super({
            getCategoryEndPoint: 'v1/categories',
            categoriesBulkActions: this.getCategoriesActions(),
            tableColumns: this.getCategoriesColumns(),
            addToBundleIdentifier: '',
            selectedCategories: [],
            categoryBreadCrumb: this.getCategoryBreadCrumbs(),
            breadCrumbStarterIcon: this.breadCrumbStarterIcon,
            paginationPerPage: 25
        });
        this.breadCrumbStarterIcon = { icon: 'pi pi-home', disabled: true };
```

Reply:
I think we can only access this parameter after calling the super.

Reply:
Still, you don't need anything dynamic to create the state, so for what I can tell everything can be init in the super.